File: iPlant.FMS.WEB/Controllers/CFG/HolidayController.cs
using iPlant.Common.Tools;
using iPlant.FMS.Models;
using iPlant.SCADA.Service;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using Microsoft.AspNetCore.Http;using Microsoft.AspNetCore.Mvc;
namespace iPlant.FMS.WEB
{
    public class HolidayController : BaseController
    {

        private static log4net.ILog logger = log4net.LogManager.GetLogger(typeof(HolidayController));
        [HttpGet]
        public ActionResult All()
        {
            Dictionary<String, Object> wResult = new Dictionary<String, Object>();
            try
            {

                

                BMSEmployee wBMSEmployee = GetSession();
                int wUserID = wBMSEmployee.ID;
                int wCompanyID = wBMSEmployee.CompanyID;

                int wWorkshopID = StringUtils.parseInt(Request.QueryParamString("WorkShopID"));

                int wYear = StringUtils.parseInt(Request.QueryParamString("year"));
                if (wYear < 2000)
                {
                    wResult = GetResult(RetCode.SERVER_CODE_ERR, RetCode.SERVER_RST_ERROR_OUT);
                    return Json(wResult);
                }

                ServiceResult<List<CFGCalendar>> wServiceResult = ServiceInstance.mCFGService.CFG_QueryHoliday(wBMSEmployee,
                        wWorkshopID, wYear);
                List<CFGCalendar> wServerRst = wServiceResult.Result;

                if (StringUtils.isEmpty(wServiceResult.getFaultCode()))
                {
                    wResult = GetResult(RetCode.SERVER_CODE_SUC, "", wServerRst, null);
                }
                else
                {
                    wResult = GetResult(RetCode.SERVER_CODE_ERR, wServiceResult.getFaultCode(), wServerRst, null);
                }
            }
            catch (Exception ex)
            {
                logger.Error(System.Reflection.MethodBase.GetCurrentMethod().Name, ex);
                wResult = GetResult(RetCode.SERVER_CODE_ERR, ex.ToString(), null, null);
            }
            return Json(wResult);
        }

        [HttpPost]

        public ActionResult Update()
        {
            Dictionary<String, Object> wResult = new Dictionary<String, Object>();
            try
            {
                Dictionary<string, object> wParam = GetInputDictionaryObject(Request);
                

                BMSEmployee wBMSEmployee = GetSession();
                int wUserID = wBMSEmployee.ID;
                int wCompanyID = wBMSEmployee.CompanyID;
                String wUserName = wBMSEmployee.Name;

                if (!wParam.ContainsKey("data") || !wParam.ContainsKey("year"))
                {
                    wResult = GetResult(RetCode.SERVER_CODE_ERR, RetCode.SERVER_RST_ERROR_OUT);
                    return Json(wResult);
                }
                int wWorkshopID = StringUtils.parseInt(wParam.ContainsKey("WorkShopID") ? wParam["WorkShopID"] : 0);
                int wYear = StringUtils.parseInt(wParam.ContainsKey("year") ? wParam["year"] : 0);

                List<CFGCalendar> wCFGCalendarList = CloneTool.CloneArray<CFGCalendar>(wParam["data"]);

                if (wCFGCalendarList != null && wCFGCalendarList.Count > 0)
                {
                    wCFGCalendarList.ForEach(p =>
                    {
                        p.OperatorID = wUserID;
                        p.OperatorName = wUserName;
                    });
                }

                ServiceResult<Int32> wServiceResult = ServiceInstance.mCFGService.CFG_RemoveYearHoliday(wBMSEmployee, wWorkshopID,
                        wYear);
                int wServerRst = wServiceResult.Result;
                if (StringUtils.isEmpty(wServiceResult.getFaultCode()))
                {
                    wServerRst = ServiceInstance.mCFGService.CFG_AddHolidayList(wBMSEmployee, wWorkshopID, wCFGCalendarList).Result;
                }

                if (StringUtils.isEmpty(wServiceResult.getFaultCode()))
                {
                    wResult = GetResult(RetCode.SERVER_CODE_SUC, "", null, wServerRst);
                }
                else
                {
                    wResult = GetResult(RetCode.SERVER_CODE_ERR, wServiceResult.getFaultCode(), null, wServerRst);
                }
            }
            catch (Exception ex)
            {
                logger.Error(System.Reflection.MethodBase.GetCurrentMethod().Name, ex);
                wResult = GetResult(RetCode.SERVER_CODE_ERR, ex.ToString(), null, null);
            }
            return Json(wResult);
        }
    }
}

File: iPlant.FMS.WEB/Controllers/CFG/HolidayController.java
package com.mes.server.controller.cfg;

import com.mes.server.controller.BaseController;
import com.mes.server.service.CFGService;
import com.mes.server.service.po.ServiceResult;
import com.mes.server.service.po.bms.BMSEmployee;
import com.mes.server.service.po.cfg.CFGCalendar;
import com.mes.server.service.utils.CloneTool;
import com.mes.server.service.utils.StringUtils;
import com.mes.server.utils.RetCode;

import java.util.List;
import java.util.Dictionary;

import javax.servlet.http.HttpServletRequest;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/Holiday")
public class HolidayController extends BaseController {

	private static Logger logger = LoggerFactory.getLogger(typeof(HolidayController));

	@Autowired
	CFGService wCFGService;

	@GetMapping("/All")
	public Object All() {
		Object wResult = new Object();
		
		return wResult;
	}

	@PostMapping("/Update")
	public Object Update() {
		Object wResult = new Object();
		try {
			if (CheckCookieEmpty()) {
				wResult = GetResult(RetCode.SERVER_CODE_UNLOGIN, "");
				return wResult;
			}

			BMSEmployee wBMSEmployee = GetSession();
			int wUserID = wBMSEmployee.ID;
			int wCompanyID = wBMSEmployee.CompanyID;
			String wUserName = wBMSEmployee.Name;

			if (!wParam.ContainsKey("data") || !wParam.ContainsKey("year")) {
				wResult = GetResult(RetCode.SERVER_CODE_ERR, RetCode.SERVER_RST_ERROR_OUT);
				return wResult;
			}
			int wWorkshopID = StringUtils.parseInt(wParam.ContainsKey("WorkShopID") ? wParam["WorkShopID"] : 0);
			int wYear = StringUtils.parseInt(wParam.ContainsKey("year") ? wParam["year"] : 0);

			List<CFGCalendar> wCFGCalendarList = CloneTool.CloneArray(wParam["data"], typeof(CFGCalendar));

			if (wCFGCalendarList != null && wCFGCalendarList.Count > 0) {
				wCFGCalendarList.forEach(p -> {
					p.OperatorID = wUserID;
					p.OperatorName = wUserName;
				});
			}

			ServiceResult<Int32> wServiceResult = wCFGService.Cfg_RemoveYearHoliday(wCompanyID, wUserID, wWorkshopID,
					wYear);
			int wServerRst = wServiceResult.Result;
			if (StringUtils.isEmpty(wServiceResult.getFaultCode())) {
				wServerRst = wCFGService.Cfg_AddHolidayList(wCompanyID, wUserID, wWorkshopID, wCFGCalendarList).Result;
			}

			if (StringUtils.isEmpty(wServiceResult.getFaultCode())) {
				wResult = GetResult(RetCode.SERVER_CODE_SUC, "", null, wServerRst);
			} else {
				wResult = GetResult(RetCode.SERVER_CODE_ERR, wServiceResult.getFaultCode(), null, wServerRst);
			}
		} catch (Exception ex) {
			logger.Error(System.Reflection.MethodBase.GetCurrentMethod().Name, ex);
			wResult = GetResult(RetCode.SERVER_CODE_ERR, ex.ToString(), null, null);
		}
		return wResult;
	}
}